### PR TITLE
8316935: [s390x] Use consistent naming for lightweight locking in MacroAssembler

### PR DESCRIPTION
**Pull request by offamitkumar**
We (s390) also needs to update our naming from fast_lock & fast_unlock to MacroAssembler::lightweight_lock and MacroAssembler::lightweight_unlock respectively.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316935](https://bugs.openjdk.org/browse/JDK-8316935): [s390x] Use consistent naming for lightweight locking in MacroAssembler (**Enhancement** - P4)

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Lutz Schmidt](https://openjdk.org/census#lucy) (@RealLucy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15915/head:pull/15915` \
`$ git checkout pull/15915`

Update a local copy of the PR: \
`$ git checkout pull/15915` \
`$ git pull https://git.openjdk.org/jdk.git pull/15915/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15915`

View PR using the GUI difftool: \
`$ git pr show -t 15915`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15915.diff">https://git.openjdk.org/jdk/pull/15915.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15915#issuecomment-1735133165)

### CODE REVIEW

**Review by TheRealMDoerr — APPROVED** [review]
Looks good and trivial.

**Comment by offamitkumar** [comment]
Thanks @TheRealMDoerr for reviewing it. It's a trivial change, Should I integrate it or wait for a another Review ?

**Comment by TheRealMDoerr** [comment]
You can integrate it after you get a 2nd review or after 24h (whatever happens first).

**Review by RealLucy — APPROVED** [review]
Looks good to me. 
Changes which are classified "trivial" by a Reviewer can be integrated with just one positive review. Now you've got both: two reviews and a trivial classification.

**Comment by offamitkumar** [comment]
> Now you've got both: two reviews and a trivial classification.

Thank you, Lutz and Martin for approving it. 

/integrate

**Pull request closed without merging**
